x/superfluid/keeper/internal/events/emit.go

```diff
@@ -0,0 +1,133 @@
+package events
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v10/x/superfluid/types"
+)
+
+func EmitSetSuperfluidAssetEvent(ctx sdk.Context, denom string, assetType types.SuperfluidAssetType) {
+	if ctx.EventManager() == nil {
+		return
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		newSetSuperfluidAssetEvent(denom, assetType),
+	})
+}
+
+func newSetSuperfluidAssetEvent(denom string, assetType types.SuperfluidAssetType) sdk.Event {
+	return sdk.NewEvent(
+		types.TypeEvtSetSuperfluidAsset,
+		sdk.NewAttribute(types.AttributeDenom, denom),
+		sdk.NewAttribute(types.AttributeSuperfluidAssetType, assetType.String()),
+	)
+}
+
+func EmitRemoveSuperfluidAsset(ctx sdk.Context, denom string) {
+	if ctx.EventManager() == nil {
+		return
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		newRemoveSuperfluidAssetEvent(denom),
+	})
+}
+
+func newRemoveSuperfluidAssetEvent(denom string) sdk.Event {
+	return sdk.NewEvent(
+		types.TypeEvtRemoveSuperfluidAsset,
+		sdk.NewAttribute(types.AttributeDenom, denom),
+	)
+}
+
+func EmitSuperfluidDelegateEvent(ctx sdk.Context, lockId uint64, valAddress string) {
+	if ctx.EventManager() == nil {
+		return
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		newSuperfluidDelegateEvent(lockId, valAddress),
+	})
+}
+
+func newSuperfluidDelegateEvent(lockId uint64, valAddress string) sdk.Event {
+	return sdk.NewEvent(
+		types.TypeEvtSuperfluidDelegate,
+		sdk.NewAttribute(types.AttributeLockId, fmt.Sprintf("%d", lockId)),
+		sdk.NewAttribute(types.AttributeValidator, valAddress),
+	)
+}
+
+func EmitSuperfluidIncreaseDelegationEvent(ctx sdk.Context, lockId uint64, amount sdk.Coins) {
+	if ctx.EventManager() == nil {
+		return
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		newSuperfluidIncreaseDelegationEvent(lockId, amount),
+	})
+}
+
+func newSuperfluidIncreaseDelegationEvent(lockId uint64, amount sdk.Coins) sdk.Event {
+	return sdk.NewEvent(
+		types.TypeEvtSuperfluidIncreaseDelegation,
+		sdk.NewAttribute(types.AttributeLockId, fmt.Sprintf("%d", lockId)),
+		sdk.NewAttribute(types.AttributeAmount, amount.String()),
+	)
+}
+
+func EmitSuperfluidUndelegateEvent(ctx sdk.Context, lockId uint64) {
+	if ctx.EventManager() == nil {
+		return
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		newSuperfluidUndelegateEvent(lockId),
+	})
+}
+
+func newSuperfluidUndelegateEvent(lockId uint64) sdk.Event {
+	return sdk.NewEvent(
+		types.TypeEvtSuperfluidUndelegate,
+		sdk.NewAttribute(types.AttributeLockId, fmt.Sprintf("%d", lockId)),
+	)
+}
+
+func EmitSuperfluidUnbondLockEvent(ctx sdk.Context, lockId uint64) {
+	if ctx.EventManager() == nil {
```

Comment:
`EventManager` should never be nil, but I guess it can't hurt. Just so many LOC :-/

Reply:
Ahh thank you. If so, I'll remove them.